extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -37,29 +39,46 @@ export abstract class AzureAuth implements vscode.Disposable {
 
 	protected readonly loginEndpointUrl: string;
 	public readonly commonTenant: Tenant;
+	public readonly organizationsTenant: Tenant;
 	protected readonly redirectUri: string;
 	protected readonly scopes: string[];
 	protected readonly scopesString: string;
 	protected readonly clientId: string;
 	protected readonly resources: Resource[];
+	private _authLibrary: AuthLibrary;
+	public get authLibrary() {
+		return this._authLibrary;
+	}
 
 
 	constructor(
 		protected readonly metadata: AzureAccountProviderMetadata,
 		protected readonly tokenCache: SimpleTokenCache,
 		protected readonly context: vscode.ExtensionContext,
+		protected clientApplication: PublicClientApplication,
 		protected readonly uriEventEmitter: vscode.EventEmitter<vscode.Uri>,
 		protected readonly authType: AzureAuthType,
 		public readonly userFriendlyName: string
 	) {
+		this._authLibrary = vscode.workspace.getConfiguration('azure').get('authenticationLibrary');
+		vscode.workspace.onDidChangeConfiguration((changeEvent) => {
+			const impactLibrary = changeEvent.affectsConfiguration('azure.authenticationLibrary');
```

Comment:
do changes to the auth library take place right away or does it require reboot?  If immediately, what about currently active connections, will they still refresh tokens correctly, etc.?

Reply:
I think we should require a reboot since there are currently two different account caches for the different libraries, so there's there's too many weird states the application could get in if we don't require a reboot.  The user will have to reconnect to any open sessions, but since we don't expect them to be changing this setting multiple times, I went ahead with this approach.